python/mxnet/gluon/nn/basic_layers.py

```diff
@@ -405,7 +405,7 @@ def __init__(self, **kwargs):
         super(Flatten, self).__init__(**kwargs)
 
     def hybrid_forward(self, F, x):
-        return x.reshape((0, -1))
+        return F.Flatten(x)
```

Comment:
This is not equivalent. https://mxnet.incubator.apache.org/versions/master/api/python/ndarray/ndarray.html?highlight=reshape#mxnet.ndarray.reshape

Reply:
@szha, thanks for the comments. 

I am not very familiar with the difference between gluon and python model.
But I see the `Flatten` is used in the AlexNext (googlenet, etc) under python example.  
https://github.com/apache/incubator-mxnet/blob/0ca6dd29cb923c59925c76981bfd5a632a72a8ca/example/image-classification/symbols/alexnet.py#L55

And other symbolic operators are used in gluon model as well, such as `BatchNorm`:
https://github.com/apache/incubator-mxnet/blob/0ca6dd29cb923c59925c76981bfd5a632a72a8ca/python/mxnet/gluon/nn/basic_layers.py#L341

So, what's functionality missing in `Flatten` operator so that it can't be used in gluon `Flatten` class?

Reply:
Non-equivalence. Please look up or try what the existing implementation does, and what Flatten op does.

Reply:
Thanks. Got it and will keep it as-is.